Fix `$openldap::server::database::limits` data type

Donatien26 opened the pull request:
fixes #345

smortex commented:
Unit tests should have caught that when adding data types…  But none tests this :cry:.

Would you mind adding a context `with all parameters set` in `spec/defines/openldap_server_database_spec.rb` that basically sets all parameters and ensure the catalog compiles with all dependenciel?

Donatien26 commented:
Yes I will try to do it !

Donatien26 commented:
i just don't know which value is expected in security

smortex commented:
This is related to olcSecurity:
https://github.com/voxpupuli/puppet-openldap/blob/master/lib/puppet/type/openldap_database.rb#L230

Maybe just set it to something like `security => { tls => 128, },`.  The idea is not to have 100% coverage, anything better than 0% is good :rofl: 

Thanks!

Donatien26 commented:
In lib/puppet/provider/openldap_database/olc.rb we have this line:

```ruby
        when %r{^olcSecurity: }
          line.split(': ')[1].split(' ').each do |variable|
            values = variable.split('=')
            security[values[0]] = values[1]
          end
        end
```
and in manifest/server/database.pp itis said that security is an Optional[String[1]], so i try to set different value like:
- "tls=128"
- "tls=128 "
- "olcsecurity: tls=128 "
but no one works, i don't know if it's a bug or if i just don't set the param correctly.

smortex commented:
> in manifest/server/database.pp itis said that security is an Optional[String[1]]

You trust the untested data types… Don't trust the untested data types :smile:

I bet you found another issue.  I tried to adjusted the code to fix this in your branch but it looks like I can't push there, so I opened #347, it passes unit tests on my machine, let's see if it pass integration tests or break something else that needs to be fixed :wink:
